File: solution/1608. Special Array With X Elements Greater Than or Equal X/1608_01.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        sort(nums.rbegin(), nums.rend());
        const int n = nums.size();
        for (int i = 0; i < n; ++i) {
            if (nums[i] < i + 1)
                return -1;
            if (i + 1 == n || nums[i + 1] < i + 1)
                return i + 1;
        }
        return -1;
    }
};

File: solution/1608. Special Array With X Elements Greater Than or Equal X/1608_01.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        n = len(nums)
        for i in range(n):
            if nums[i] < i + 1:
                break
            if i + 1 == n or nums[i + 1] < i + 1:
                return i + 1
        return -1
